Makefile

```diff
@@ -0,0 +1,33 @@
+.DEFAULT_GOAL := help
```

Comment:
Adding a Makefile here is a bit sketchy, but technically avoids stomping on the CMake and generated `build/Makefile`

I think it's nice to [capture some of the test instructions from here](https://osquery.readthedocs.io/en/stable/development/building/#testing), and I leverage this in the setup of the Devcontainer.